python/lsst/summit/utils/utils.py

```diff
@@ -297,6 +299,80 @@ def detectObjectsInExp(exp, nSigma=10, nPixMin=10, grow=0):
     return footPrintSet
 
 
+def fluxesFromFootprints(footprints, parentImage, subtractImageMedian=False):
+    """Calculate the flux from a set of footprints, given the parent image,
+    optionally subtracting the whole-image median from each pixel as a very
+    rough background subtraction.
+
+    Parameters
+    ----------
+    footprints : `lsst.afw.detection.FootprintSet` or
+                 `lsst.afw.detection.Footprint` or
+                 `iterable` of `lsst.afw.detection.Footprint`
+        The footprints to measure.
+    parentImage : `lsst.afw.image.Exposure`
+        The parent exposure.
+    subtractImageMedian : `bool`, optional
+        Subtract a whole-image median from each pixel in the footprint when
+        summing as a very crude background subtraction. Does not change the
+        original image.
+
+    Returns
+    -------
+    fluxes : `list` of `float`
+        The fluxes for each footprint.
+
+    Raises
+    ------
+    ValueError : raise for unsupported types.
+    """
+    median = 0
+    if subtractImageMedian:
+        median = np.nanmedian(parentImage.image.array)
+
+    # poor person's single dispatch
+    badTypeMsg = ("This function works with FootprintSets, single Footprints, and iterables of Footprints. "
+                  f"Got {type(footprints)}: {footprints}")
+    if isinstance(footprints, FootprintSet):
+        fps = footprints.getFootprints()
```

Comment:
I'm not keen on `fps` here, but you've already got `footprints` as the arg name. It is an `if:elif:` block, so you could just re-assign to `footprints` in the places you needed to?

Reply:
Yup, sounds good, thanks.